test/stores/VDocument.test.ts

```diff
@@ -352,19 +352,968 @@ describe('stores', () => {
+                        // A <br> is automatically added to make the <p>
+                        // visible.
+                        contentAfter: '<p>[]<br></p>',
+                    });
+                    testEditor({
+                        contentBefore: '<p>[<br>]</p>',
+                        stepFunction: deleteForward,
+                        // The <br> is there only to make the <p> visible.
+                        // It does not exist in VDocument and selecting it
+                        // has no meaning in the DOM.
+                        contentAfter: '<p>[]<br></p>',
+                    });
+                    testEditor({
+                        contentBefore: '<p>abc[]</p>',
+                        stepFunction: deleteForward,
+                        contentAfter: '<p>abc[]</p>',
+                    });
+                });
+                it('should delete the first character in a paragraph', () => {
+                    testEditor({
+                        contentBefore: '<p>[]abc</p>',
+                        stepFunction: deleteForward,
+                        contentAfter: '<p>[]bc</p>',
+                    });
+                });
+                it('should delete a character within a paragraph', () => {
+                    testEditor({
+                        contentBefore: '<p>a[]bc</p>',
+                        stepFunction: deleteForward,
+                        contentAfter: '<p>a[]c</p>',
+                    });
+                });
+                it('should delete the last character in a paragraph', () => {
+                    testEditor({
+                        contentBefore: '<p>ab[]c</p>',
+                        stepFunction: deleteForward,
+                        contentAfter: '<p>ab[]</p>',
+                    });
+                    testEditor({
+                        contentBefore: '<p>ab []c</p>',
+                        stepFunction: deleteForward,
+                        // The space should be converted to an unbreakable space
+                        // so it is visible.
+                        contentAfter: '<p>ab&nbsp;[]</p>',
+                    });
+                });
+                it('should merge a paragraph into an empty paragraph', () => {
+                    testEditor({
+                        contentBefore: '<p>[]<br></p><p>abc</p>',
+                        stepFunction: deleteForward,
+                        contentAfter: '<p>[]abc</p>',
+                    });
+                });
+                it('should delete a character after a trailing space', () => {
+                    testEditor({
+                        contentBefore: '<p>abc []d</p>',
+                        stepFunction: deleteForward,
+                        // The space should be converted to an unbreakable space
+                        // so it is visible.
+                        contentAfter: '<p>abc&nbsp;[]</p>',
+                    });
+                });
```

Comment:
Oops. Remove this test: the exact same one is within 'should delete the last character in a paragraph'.